src/tests/test_kube_api.py

```diff
@@ -127,27 +126,21 @@ def kube_api_test(
         infra_env.create(
             cluster_deployment, secret, proxy, ignition_config_override, ssh_pub_key=cluster_config.ssh_public_key
         )
+
         agents = self.start_nodes(nodes, infra_env, cluster_config)
 
-        log.info("Waiting for agent status verification")
-        Agent.wait_for_agents_to_install(agents)
+        if len(nodes) == 1:
+            self._set_agent_cluster_install_machine_cidr(agent_cluster_install, nodes)
```

Comment:
In order to simulate Git-Ops as good as we can, we should not patch k8s objects but rather have all objects set only once
Can we do that throughout the whole test? (also applies to ``set_api_vip`` and ``set_ingress_vip`` at the bottom)
Otherwise we might evade some pretty problematic bugs

Reply:
@ori-amizur can we change the flow of this test so the VIP will be calculated before creating the aci? Does it requires staring the nodes?

Reply:
No. You can set the VIPs in the ACI spec